keps/sig-network/20200130-Uncapped-IPv6.md

```diff
@@ -0,0 +1,316 @@
+---
+title: Uncapped IPv6
+authors:
+  - "@aojea"
+owning-sig: sig-network
+participating-sigs:
+  - sig-scalability
+  - sig-api-machinery
+reviewers:
+  - "@thockin"
+  - "@bowei"
+  - "@liggitt"
+  - "@smarterclayton"
+  - "@wojtek-t"
+approvers:
+  - "@thockin"
+  - "@liggitt"
+  - "@wojtek-t"
+editor: TBD
+creation-date: 2020-01-30
+last-updated: 2020-01-30
+status: provisional
+see-also:
+  - "/keps/sig-network/20190714-IPv6-beta-proposal.md"
+  - "https://github.com/kubernetes/kubernetes/issues/44918"
+  - "https://github.com/kubernetes/kubernetes/pull/86620"
+  - "https://github.com/kubernetes/kubernetes/pull/79993"
+
+---
+
+# Remove IPv6 Allocation Restrictions
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Roaring bitmaps](#roaring-bitmaps)
+  - [Services IP Allocator](#services-ip-allocator)
+  - [Node IPAM Controller and Range Allocator](#node-ipam-controller-and-range-allocator)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Kubernetes impose some restrictions on the number of Service IPs and CIDR Range sizes when using IPv6.
+This is done to avoid scalability issues caused by IPv6, we should take into account that IPv4 has an
+address namespace of 2^32 addresses vs 2^128 with IPv6.
+
+## Motivation
+
+One of the main motivation to restric allocations when using IPv6 is caused because Kubernetes is using internally bitmaps (or bitsets) to store the data.
```

Comment:
nit:
```suggestion
One of the main motivation to restrict allocations when using IPv6 is caused because Kubernetes is using internally bitmaps (or bitsets) to store the data.
```